ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -99,9 +99,68 @@ struct IntegrationPointData final
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW;
 };
 
+/// Used by for extrapolation of the integration point values. It is ordered
+/// (and stored) by integration points.
+template <typename ShapeMatrixType>
+struct SecondaryData
+{
+    std::vector<ShapeMatrixType, Eigen::aligned_allocator<ShapeMatrixType>> N;
```

Comment:
IMHO there should be some kind of indication that N is for the displacement and not for the pressure variable. E.g., rename to `N_u`